### PR TITLE
Rename everything to metals

### PR DESCRIPTION
**Pull request by gabro**
I'm sure I've missed something 😅

~I'll update this once #195 is in.~ Done!

### CODE REVIEW

**Comment by gabro** [comment]
Mmm one test is consistently failing (I've restarted the build once) and I don't know why:

```
X tests.search.SymbolIndexTest.references.a.a.<<x>> 4ms 
  utest.AssertionError: assert(missingLocations.isEmpty)
missingLocations: scala.collection.immutable.Set[org.langmeta.lsp.Location] = Set(Location(file:/hom
e/travis/build/scalameta/metals/test-workspace/src/main/scala/example/User.scala,Range(Position(6,15
),Position(6,16))))
  utest.asserts.Asserts$.assertImpl(Asserts.scala:114)
  tests.search.SymbolIndexTest$.assertSymbolReferences$1(SymbolIndexTest.scala:133)
  tests.search.SymbolIndexTest$.$anonfun$tests$22(SymbolIndexTest.scala:201)
```

**Comment by gabro** [comment]
Ah, I think I've got it. It was a minor edit to the test-workspace that I pushed by mistake. Let's see if the CI is happy now.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -8,8 +8,8 @@ Please read them for information about how to create good bug reports and submit
 
 
 ## Project structure
-- `languageserver` contains a Scala implementation of the [Language Server Protocol specification](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md) (this is a fork of https://github.com/dragos/dragos-vscode-scala)
-- `metaserver` contains a Scala implementation of a language server based on Scalameta, which uses `languageserver`
+- `lsp4s` contains a Scala implementation of the [Language Server Protocol specification](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md) (this is a fork of https://github.com/dragos/dragos-vscode-scala)
```
Comment (olafurpg) [inline CONTRIBUTING.md]: unrelated to this PR, but guess lsp4s is barely a fork anymore

File: build.sbt [hunk 2]
```diff
@@ -127,7 +127,7 @@ lazy val lsp4s = project
   .settings(crossScalaVersions := List(V.scala211, V.scala212))
   .dependsOn(jsonrpc)
```
Comment (olafurpg) [inline build.sbt]: Can we set `name := "metalsRoot"` in the top-level? By default sbt names the root project after the directory name, unless there is a project with a conflicting in which case it creates a bogus name like "default-fd2bf4". This name appears in IntelliJ after importing the build so I think "metalsRoot" would be nice

File: metals/src/main/scala/scala/meta/metals/ScalametaServices.scala [hunk 3]
```diff
@@ -500,10 +500,10 @@ class ScalametaServices(
   }
 }
 
-object ScalametaServices extends LazyLogging {
+object MetalsServices extends LazyLogging {
   lazy val cacheDirectory: AbsolutePath = {
     val path = AbsolutePath(
-      ProjectDirectories.fromProjectName("metaserver").projectCacheDir
```
Comment (olafurpg) [inline metals/src/main/scala/scala/meta/metals/ScalametaServices.scala]: PSA: remove that `metaserver` cache directory y'all 😄

File: build.sbt (hunk 2 above)
Comment (gabro) [reply build.sbt]: 👍 done. Thanks for explaining where the bogus name came from, also the sbt launcher displays it and I was confused.

**Pull request merged**